integrations/hubspot/oauth.go

```diff
@@ -58,5 +63,42 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Sanity check: the connection ID is valid.
+	cid, err := sdktypes.StrictParseConnectionID(c.ConnectionID)
+	if err != nil {
+		l.Warn("save connection: invalid connection ID", zap.Error(err))
+		c.AbortBadRequest("invalid connection ID")
+		return
+	}
+
+	vsid := sdktypes.NewVarScopeID(cid)
+	common.SaveAuthType(r, h.vars, vsid)
+
+	// Get portal ID
+	accountURL := "https://api.hubapi.com/integrations/v1/me"
+	accountResp, err := common.HTTPGet(r.Context(), accountURL, auth)
+	if err != nil {
+		l.Warn("failed to get HubSpot account details", zap.Error(err))
+		c.AbortServerError("failed to get account details")
```

Comment:
i don't see a return here? this will still progress to finalize. 
also - when you put the return there is no need for the else afterwards, you can deindent that code.